aws/resource_aws_networkfirewall_firewall.go

```diff
@@ -50,6 +50,42 @@ func resourceAwsNetworkFirewallFirewall() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
+			"firewall_status": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"sync_states": {
+							Type:     schema.TypeList,
```

Comment:
Ahh yess forgot to comment here explanation for using TypeList 😄 sooo when running the acctests, the typeset checks failed stating that the `sync_states` didn't have the correct values in the case of resource Updates e.g. when removing one subnet for a new subnet. i'll run the tests again to make sure it wasn't a fluke tho.. i figured it was one of those quirky TypeSet behaviors but I could be wrong

Reply:
We may need to tell the resource that the `firewall_status` attribute needs to be recomputed on updates to those other arguments, which would have been a problem anyways. Practitioners otherwise might need to double apply to see downstream changes. Something like the below might get the implementation started:

```go
CustomizeDiff: customdiff.Sequence(
	customdiff.ComputedIf("firewall_status", func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) bool {
		return diff.HasChange("subnet_mapping")
	}),
),
```

Reply:
didn't think about that! always forget about these cool hacks